### PR TITLE
Feat: Added option to disable SSL verify

### PR DESCRIPTION
**Pull request by maayanbar13**
# Pull Request Check List
Resolves: #1556
 
* [x]  Added **tests** for changed code.
* [x]  Updated **documentation** for changed code.

 I'm opening this new pull request to replace [/pull/2912](https://github.com/python-poetry/poetry/pull/2912) since @Celeborn2BeAlive is no longer following the issue.

### CODE REVIEW

**Comment by SergioBlnc** [comment]
Hi,

When this feat will be released?

Thanks in advance
Sbg

**Comment by mcdevitts** [comment]
Being able to disable SSL verification is whats keeping me from using poetry since we have a private repository. What can I do to help get this pull request merged?

File: poetry/factory.py [hunk 1]
```diff
@@ -146,6 +146,7 @@ def create_legacy_repository(
         from .repositories.legacy_repository import LegacyRepository
         from .utils.helpers import get_cert
         from .utils.helpers import get_client_cert
+        from .utils.helpers import get_truested
```
Comment (therewillbeblood) [inline poetry/factory.py:149]:
```suggestion
        from .utils.helpers import get_trusted
```

File: poetry/factory.py [hunk 2]
```diff
@@ -163,4 +164,5 @@ def create_legacy_repository(
             config=auth_config,
             cert=get_cert(auth_config, name),
             client_cert=get_client_cert(auth_config, name),
+            trusted=source.get("trusted", False) or get_truested(auth_config, name),
```
Comment (therewillbeblood) [inline poetry/factory.py:167]:
```suggestion
            trusted=source.get("trusted", False) or get_trusted(auth_config, name),
```

File: poetry/utils/helpers.py [hunk 3]
```diff
@@ -71,6 +71,14 @@ def get_client_cert(config: Config, repository_name: str) -> Optional[Path]:
         return None
 
 
+def get_truested(config: Config, repository_name: str) -> Optional[bool]:
```
Comment (therewillbeblood) [inline poetry/utils/helpers.py:74]:
```suggestion
def get_trusted(config: Config, repository_name: str) -> Optional[bool]:
```

File: tests/utils/test_helpers.py [hunk 4]
```diff
@@ -3,6 +3,7 @@
 from poetry.core.utils.helpers import parse_requires
 from poetry.utils.helpers import get_cert
 from poetry.utils.helpers import get_client_cert
+from poetry.utils.helpers import get_truested
```
Comment (therewillbeblood) [inline tests/utils/test_helpers.py:6]:
```suggestion
from poetry.utils.helpers import get_trusted
```

**Review by therewillbeblood — CHANGES_REQUESTED** [review]
what about fixing the typos in get_trusted right away?

File: tests/utils/test_helpers.py [hunk 5]
```diff
@@ -70,3 +71,10 @@ def test_get_client_cert(config):
     config.merge({"certificates": {"foo": {"client-cert": client_cert}}})
 
     assert get_client_cert(config, "foo") == Path(client_cert)
+
+
+def test_get_trusted(config):
+    trusted = "true"
+    config.merge({"certificates": {"foo": {"trusted": trusted}}})
+
+    assert get_truested(config, "foo")
```
Comment (therewillbeblood) [inline tests/utils/test_helpers.py:80]: there seems to be a typo here as well: get_trusted

Comment (maayanbar13) [reply tests/utils/test_helpers.py]: Fixed :)

Comment (therewillbeblood) [reply tests/utils/test_helpers.py]: Thanks! However it seems more must be done as part of this change:
` RuntimeError: The Poetry configuration is invalid:
E             - [source.0] Additional properties are not allowed ('trusted' was unexpected)
`

Comment (maayanbar13) [reply tests/utils/test_helpers.py]: This is expected, as written in the original post, in order for tests to pass you must first complete the mentioned PR in `poetry-core` as it contains the required changes to the poetry configuration schema.

**Comment by maayanbar13** [comment]
I've Reverted the changes according to [python-poetry/poetry-core#80](https://github.com/python-poetry/poetry-core/pull/80). Now disabling SSL verify is only through poetry config certificates.foo.verify False

**Comment by McClunatic** [comment]
If I can help get this merged, please let me know! I'm unable to use Poetry at work without it.

**Comment by Never77** [comment]
> If I can help get this merged, please let me know! I'm unable to use Poetry at work without it.

have the same trouble, if it can help to merge, i'm available to help. I need this feature to use private repository at the office.

**Review by Never77 — APPROVED** [review]

**Comment by abn** [comment]
@maayanbar13 thanks for working on this. As we are slowly improving how the http interactions are managed, I'd expect this to go via the authenticator now rather than being configured at the repository instance level. We want to make sure that the requests going to this url is also trusted, similar to how auth and certs are applied. The repository instances can expose this property similar to authenticated url property.

File: src/poetry/installation/pip_installer.py [hunk 6]
```diff
@@ -56,6 +56,8 @@ def install(self, package: Package, update: bool = False) -> None:
                     f" {parsed.hostname}</warning>"
                 )
                 args += ["--trusted-host", parsed.hostname]
+            elif repository.trusted:
+                args += ["--trusted-host", parsed.hostname]
```
Comment (abn) [inline src/poetry/installation/pip_installer.py:60]: This only affects the old installer as the new installer downloads the packages via the authenticator.

**Comment by pabloh007** [comment]
Let me know when this will be released  and if there is anything I can help with to move this along.

**Comment by neersighted** [comment]
> Let me know when this will be released and if there is anything I can help with to move this along.

Please refer to the [above comment](https://github.com/python-poetry/poetry/pull/3676#issuecomment-1120268812) -- this is unable to be merged as it, but recent refactors have made possible a more idiomatic and clean way to introduce this functionality.

**Comment by abn** [comment]
@maayanbar13 @neersighted  I have rolled up the functional change requested here with a few other structural changes to cert handling in #5719.

**Pull request closed without merging**